### PR TITLE
refactor!: update auth-service to latest thirdparty api definition

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

File: migrations/20220322001917_address.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ --------------
+ ******/
+import { Knex } from 'knex'
+
+export async function up (knex: Knex): Promise<void | Knex.SchemaBuilder> {
+  return knex.schema.hasTable('Scope')
+    .then((exists: boolean): Knex.SchemaBuilder | void => {
+      if (exists) {
+        return knex.schema.alterTable('Scope', function (table) {
+          table.renameColumn('address', 'address')
```
Comment (lewisdaly) [inline migrations/20220322001917_address.ts:32]: What is this migration doing?

File: migrations/20220322001917_address.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        })
+      }
+    })
+}
+
+export function down (knex: Knex): Promise<void | Knex.SchemaBuilder> {
+  return knex.schema.hasTable('Scope')
```
Comment (lewisdaly) [inline migrations/20220322001917_address.ts:39]: This is meant to reverse whatever the migration was, but I'm not sure what that is, so I can't tell if this is correct

File: migrations/20220322001917_address.ts (hunk 1 above)
Comment (kleyow) [reply migrations/20220322001917_address.ts:32]: This is ment to rename accountId to address. I must be sleep deprived.

File: test/unit/domain/stateMachine/verifyTransaction.model.test.ts [hunk 3]
```diff
@@ -25,6 +25,17 @@ optionally within square brackets <email>.
...
 
+/* IMPORTANT
+   The fido challenges found in `auth-service` are signed with
+   Kevin Leyow's <kevin.leyow@modusbox.com> Yubikey. If the POST /consent
+   `consentId` and `scopes` ever change form you will need to derivie and resign the challenges,
+   update the `credential` object and update this PSA.
+   You will also need to update the public keys found in every verify transaction flow.
+   Use https://github.com/mojaloop/contrib-fido-test-ui to retrieve data used to update
+   the response bodies.
+*/
```
Comment (elnyry-sam-k) [inline test/unit/domain/stateMachine/verifyTransaction.model.test.ts:36]: Maybe add a link to the steps for that from here: https://github.com/mojaloop/contrib-fido-test-ui

File: test/unit/server/handlers/thirdpartyRequestsVerifications.test.ts [hunk 4]
```diff
@@ -49,13 +49,13 @@ const PostThirdpartyRequestsVerificationsRequest = {
   },
   params: {},
   payload: {
-    //TODO:
+    // TODO:
```
Comment (elnyry-sam-k) [inline test/unit/server/handlers/thirdpartyRequestsVerifications.test.ts:52]: TODO what? Either we need a follow-up story to do whats needed here or remove this

File: src/domain/challenge.ts [hunk 5]
```diff
@@ -74,6 +74,13 @@ export function deriveChallenge (consentsPostRequest: tpAPI.Schemas.ConsentsPost
     scopes: consentsPostRequest.scopes
   }
 
+  rawChallenge.scopes = rawChallenge.scopes.map((s) => ({
+    address: s.address,
+    actions: s.actions
+  }))
+
   const RFC8785String = canonicalize(rawChallenge)
+  console.log(RFC8785String)
```
Comment (elnyry-sam-k) [inline src/domain/challenge.ts:83]: debug maybe?

File: test/unit/domain/stateMachine/verifyTransaction.model.test.ts (hunk 3 above)
Comment (kleyow) [reply test/unit/domain/stateMachine/verifyTransaction.model.test.ts:36]: done

File: test/unit/server/handlers/thirdpartyRequestsVerifications.test.ts (hunk 4 above)
Comment (kleyow) [reply test/unit/server/handlers/thirdpartyRequestsVerifications.test.ts]: removed.

File: src/domain/challenge.ts (hunk 5 above)
Comment (kleyow) [reply src/domain/challenge.ts]: thanks! removed.

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**